### ISSUE TITLE
this.context.listView.columns is an empty array

### ISSUE THREAD

**Issue by florianwachter** [SharePoint/sp-dev-docs#5681]
## Category

- [x] Bug

## Describe the bug

this.context.listView.columns in a Command Set is always an empty array. This has been working fine until today.

## Steps to reproduce

Create a new command set extension and log this.context.listView.columns.

## Expected behavior

this.context.listView.columns in Command Set provides a list columns in the current view

## Environment details (*development & target environment*)
Office 365
The issue occurs in target and standard release.

**Comment by OS-Lee** [comment]
Hi,

I debug in my environment, I could get it in OnExecute while not in onListViewUpdated.

![image](https://user-images.githubusercontent.com/40845109/81367712-900d6780-9120-11ea-8989-63ccfb1f2369.png)

**Lee
Microsoft SharePoint Community Support**

**Comment by florianwachter** [comment]
Thanks Lee. OnExecute is too late for me. Until yesterday, this worked perfectly in OnListViewUpdated.
Hundreds of customers use this app and suddenly it stops working...

**Comment by 69mustang302** [comment]
At our company we have the excact same issue in a commandbar extension.
In the ` onInit`  of the commandset class:

```typescript
export default class KpiItemEditorCommandSet extends BaseListViewCommandSet<ICompanyCommandSetProperties> 
[...]
@override
public async onInit(): Promise<void>
{
   const columns = this.context.listView.columns
}
```

This suddenly returns an emprty array, while functionality in our case depends on it, while it has worked just a couple of days ago.
Our company is plus-size, and more than a thousend people depend on this.
Also ` onExcecute`  is too late for us, we need to decide wheter to show the command or not, this is only possible in ` onInit` 

Please let us know what the solution is to fix this.

**Comment by 69mustang302** [comment]
@florianwachter have you constructed a workaround?

**Comment by florianwachter** [comment]
@69mustang302: I now use the viewId from this.context.pageContext.legacyPageContext.viewId to get the columns using the REST api instead. Hope this helps.

**Comment by 69mustang302** [comment]
@florianwachter thank you, that helps! As a workaround i indeed intend to make a rest query to figure out wheter or not to show my commandbuttons. I hoped for another alternative though, since this is far from ideal because this code will run on any lists in your site, and this will lead to unnecessary traffic and performance impact on lists. So lets hope for a quick resolution by @andrewconnell

**Comment by andrewconnell** [comment]
@69mustang302 why me?

**Comment by 69mustang302** [comment]
@andrewconnell ha, because you added the labels, i assumed you could help ;)

**Comment by andrewconnell** [comment]
@69mustang302 :) I help manage/triage the list. SPFx is my domain & I answer questions where I can, but I can't "fix" anything. I don't work for Microsoft... this is one of my community contributions and giving back to the SP dev community.

**Comment by 69mustang302** [comment]
Thanks for helping, ofcourse it's greatly appreciated!

**Comment by patmill** [comment]
Missed this due to some metadata hiccups.  This has been fixed (it was a duplicate of #867 )
